src/migration/utilities.js

```diff
@@ -1,14 +1,32 @@
-function getDependency(artifacts, network, deployer, path) {
+function initializeContract(path, deployer, accounts) {
+  const contract = require("truffle-contract")
+
+  const Contract = contract(require(path))
+  Contract.setProvider(deployer.provider)
+  Contract.setNetwork(deployer.network_id)
+  // For some reason the automatic value calculation is not working, hence we do:
+  Contract.defaults({
+    from: accounts[0],
+    gas: 8e6,
+  })
+  return Contract
+}
+
+function getDependency(artifacts, network, deployer, accounts, path) {
   let Contract
 
   if (isDevelopmentNetwork(network)) {
-    Contract = artifacts.require(path)
+    // If this migration script is used from the repository dex-contracts, the contract
+    // data is received via the artificats.require.
+    // If this migration script is used from an external project, the first try statement
+    // will fail and it will get the contracts from the function initializeContract.
+    try {
+      Contract = artifacts.require(path.split("/").pop())
```

Comment:
Would it be possible to always load the resource from disk? What's the advantage of using the artefact when we are on development?

Reply:
If this script should be used in dex-contracts, then we don't have the artifact in the path provided. Then we need to get it out of the build folder.

Reply:
I'm wondering when we deploy, will the artifact networks get updated when we don't use `artifacts`?